c2rust-analyze/src/util.rs

```diff
@@ -2,11 +2,40 @@ use crate::labeled_ty::LabeledTy;
 use rustc_hir::def::DefKind;
 use rustc_hir::def_id::DefId;
 use rustc_middle::mir::{
-    Field, Local, Mutability, Operand, PlaceElem, PlaceRef, ProjectionElem, Rvalue,
+    Field, Local, Mutability, Operand, Place, PlaceElem, PlaceRef, ProjectionElem, Rvalue,
 };
 use rustc_middle::ty::{AdtDef, DefIdTree, SubstsRef, Ty, TyCtxt, TyKind, UintTy};
+use std::collections::HashMap;
 use std::fmt::Debug;
 
+/// A mapping for substituting [`Place`]s adhering to the
+/// following pattern
+/// ```mir
+/// _1 = malloc(...);
+/// _2 = _1 as *mut T;
+/// ```
+///
+/// In this case, `_1` would be mapped to `_2`, which is indicative
+/// of the amended statement:
+/// ```
+/// _2 = malloc(...);
+/// ```
```

Comment:
```suggestion
/// A mapping for substituting [`Place`]s adhering to the
/// following pattern
///
/// ```mir
/// _1 = malloc(...);
/// _2 = _1 as *mut T;
/// ```
///
/// where `_1` is [`*c_void`](core::ffi::c_void).
///
/// In this case, `_1` would be mapped to `_2`, which is indicative
/// of the amended statement:
///
/// ```mir
/// _2 = malloc(...);
/// ```
```

Reply:
Same question as https://github.com/immunant/c2rust/pull/788#discussion_r1102108079